src/main/java/org/sagebionetworks/bridge/workerPlatform/bridge/PagedResourceIterator.java

```diff
@@ -1,47 +1,106 @@
 package org.sagebionetworks.bridge.workerPlatform.bridge;
 
+import java.io.IOException;
 import java.util.Iterator;
 import java.util.List;
-import java.util.function.Function;
+import java.util.NoSuchElementException;
+
+import org.sagebionetworks.bridge.rest.exceptions.BridgeSDKException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
 
 /**
  * Iterate over paged APIs so we don't have to load all items into memory, which some methods in
  * the BridgeHelper are doing. Particularly for participants, this is likely to consume a lot of
- * memory.
+ * memory. This iterator stops iteration if it encounters an error from the server (rather than
+ * retrying or proceeding to the next page...this behavior may be changed in the future).
  */
 public class PagedResourceIterator<T> implements Iterator<T> {
+    private static final Logger LOG = LoggerFactory.getLogger(PagedResourceIterator.class);
     
-    private final Function<Integer, List<T>> func;
+    private static final int NUMBER_RETRIES = 3;
+    
+    @FunctionalInterface
+    public interface IOFunction<S, T, R> {
+        R apply(S s, T t) throws IOException;
+    }
+    
+    private final IOFunction<Integer, Integer, List<T>> func;
     private final int pageSize;
     
     private List<T> page;
     private int nextIndex;
     private int offsetBy;
+    private int retryDelayInSeconds;
     
-    public PagedResourceIterator(Function<Integer, List<T>> func, int pageSize) {
+    public PagedResourceIterator(IOFunction<Integer, Integer, List<T>> func, int pageSize, int retryDelayInSeconds) {
         this.func = func;
         this.pageSize = pageSize;
+        this.retryDelayInSeconds = retryDelayInSeconds;
+        loadNextPage();
+    }
+    
+    public PagedResourceIterator(IOFunction<Integer, Integer, List<T>> func, int pageSize) {
+        this(func, pageSize, 5);
     }
     
-    private void getNextPage() {
-        page = func.apply(offsetBy);
+    /**
+     * If the iterator encounters a potentially transient error, including Request
+     * Timeout (408), Too Many Requests (429), Bad Gateway (502), Service
+     * Unavailable (503), or Gateway Timeout (504), it will retry up to three times
+     * with exponential back-off. Other exceptions cause the iterator to stop iteration.
+     */
+    private void loadNextPage() {
+        page = ImmutableList.of(); // a failure will stop iteration.
+        for (int i=0; i <= NUMBER_RETRIES; i++) {
+            if (i > 0) { // this is a retry
+                try {
+                    long timeout = (long)Math.pow(retryDelayInSeconds, i);
```

Comment:
With the default retry delay of 5, this means that the first retry delay will be 5 seconds, then 25 seconds, then 125 seconds. This by itself isn't a problem, but it's essentially super-exponential backoff.

A more traditional approach would be 5 seconds, then 10 seconds, then 20 seconds, or retryDelayInSeconds*Math.pow(2, i);